### ISSUE TITLE
class-id mapping for mapillary dataset

### ISSUE THREAD

**Issue by 0liliulei** [openseg-group/openseg.pytorch#59]
I cannot find any class-id mapping in README or the config file. Just like Road in the ground truth with a label of 0 and traffic light is 1, the unlabeled is 255, etc.

could you provide the mapping for v1.2 of mapillary?

**Comment by hsfzxjy** [comment]
Actually mapillary has 65 classes. You can find the mapping here.

**Comment by 0liliulei** [comment]
@hsfzxjy in this file [https://github.com/openseg-group/openseg.pytorch/blob/2c459f3b42deee26194f1802f353887d945e14c4/lib/datasets/preprocess/mapillary/mapillary_generator.py#L71](url) 
you say road is 7 and sidewalk is 11 in the ground truth, but how can I know what is the number of `traffic light ` or `bird`, what is the number of unlabeled things.

**Comment by hsfzxjy** [comment]
> @hsfzxjy in this [file](https://github.com/openseg-group/openseg.pytorch/blob/2c459f3b42deee26194f1802f353887d945e14c4/lib/datasets/preprocess/mapillary/mapillary_generator.py#L71)
> you say road is 7 and sidewalk is 11 in the ground truth, but how can I know what is the number of `traffic light ` or `bird`, what is the number of unlabeled things.

The file is for some old purpose and is not used now. This file was for generating 19-class Mapillary labels, which maps the original 65 classes into 19 classes compatible with Cityscapes. However for OCR Mapillary pretraining, we use the original 65-class labels.

To train on Mapillary, you can simply place the _original_ images and labels into following structures like other datasets:
```
mapillary/
    train/
        image/
        label/
    val/
        image/
        label/
```

**Comment by 0liliulei** [comment]
@hsfzxjy  Thank you for your quickly reply. I can recognize that this 19-classes is the same as cityscape. Does pretrain on only 19 classes performs worse than on 65 classes when applying to fine-tune cityscape?

**Comment by 0liliulei** [comment]
the ignore index for mapillary is -1, which means that unlabeled is -1, am i right?

**Comment by hsfzxjy** [comment]
> @hsfzxjy Thank you for your quickly reply. I can recognize that this 19-classes is the same as cityscape. Does pretrain on only 19 classes performs worse than on 65 classes when applying to fine-tune cityscape?

I can remember early experiments revealed that training on 65-classes performed better.



> the ignore index for mapillary is -1, which means that unlabeled is -1, am i right?

Correct.
